### PR TITLE
Fix issue with specifying release notes

### PR DESCRIPTION
**Pull request by Kumuluzz**
Fixed a bug where it wasn't possible to submit release notes.

Unfortunately this solution doesn't handle \' very elegantly if it's written in the release notes.

### CODE REVIEW

File: lib/fastlane/plugin/tpa/actions/tpa_action.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@ def self.upload_options(params)
         options << "-F publish=#{params[:publish]}"
 
         if params[:notes]
-          options << "-F notes=#{params[:notes]}"
+          options << "-F notes='#{params[:notes].gsub("'", "")}'"
```
[the remark this answers is not in the document]
Comment (Kumuluzz) [reply lib/fastlane/plugin/tpa/actions/tpa_action.rb]: @kringelbach jep, unfortunately it doesn't work. It throws the following error when uploading:

```
sh: -c: line 0: unexpected EOF while looking for matching `''
sh: -c: line 1: syntax error: unexpected end of file
```

**Comment by Kumuluzz** [comment]
Updated the `tpa` action to be based on the `RestClient` in order to gracefully handle release notes with either `"` or `'` inside of them.

This means that the progress bar has been deprecated. Since the official `upload_to_testflight` action doesn't show a progress bar neither, then I believe this is an okay solution.

The status messages written in the console to the user matches the one in `upload_to_testflight`:
https://github.com/fastlane/fastlane/blob/1af33c0884e783867bdc07a324280ec6bef124ad/fastlane_core/lib/fastlane_core/itunes_transporter.rb#L371

**Comment by mbogh** [comment]
Perhaps rename `tpa` to `upload_to_tpa` or make it an alias.

It reads much better in your `Fastfile`.

File: lib/fastlane/plugin/tpa/actions/tpa_action.rb [hunk 2]
```diff
@@ -14,17 +14,12 @@ def self.run(params)
...
-          if res.key?("detail")
-            raise "Something went wrong while uploading your app to TPA: #{res['detail']}"
-          else
-            raise "Something went wrong while uploading your app to TPA: #{ex.response}"
-          end
+          handle_exception_response(ex)
         rescue => ex
-          UI.error('Something went wrong while uploading your app to TPA')
-          raise ex
+          UI.crash!("4Something went wrong while uploading your app to TPA: #{ex}")
```
Comment (bjarkehs) [inline lib/fastlane/plugin/tpa/actions/tpa_action.rb]: 4Something?

Comment (Kumuluzz) [reply lib/fastlane/plugin/tpa/actions/tpa_action.rb]: Typo 😊

**Pull request merged**